### PR TITLE
Adjoint refinement options input

### PR DESCRIPTION
**Pull request by vikramvgarg**
Changes to allow user to request an Adjoint Refinement Error Estimate for a QoI. 

Passes make check and tested with parsed laplace regression. Core capability verified earlier in libMesh.

### CODE REVIEW

File: src/solver/include/grins/mesh_adaptive_solver_base.h [hunk 1]
```diff
@@ -72,6 +72,7 @@ namespace GRINS
     libMesh::Real _refine_fraction;
     libMesh::Real _coarsen_fraction;
     libMesh::Real _coarsen_threshold;
+    bool compute_QoI_error_estimate;
```
Comment (pbauman) [inline src/solver/include/grins/mesh_adaptive_solver_base.h]: Please use consistent naming. For members of a class, the name is prefixed with a `_` and all the characters are lower case.

File: src/solver/src/steady_mesh_adaptive_solver.C [hunk 2]
```diff
@@ -110,6 +112,28 @@ namespace GRINS
...
+	      libMesh::AdjointRefinementEstimator* adjoint_ref_error_estimator = libMesh::libmesh_cast_ptr<libMesh::AdjointRefinementEstimator*>( context.error_estimator.get() );
+	      std::cout<<"The error estimate for QoI("<<i<<") is: "<<adjoint_ref_error_estimator->get_global_QoI_error_estimate(i)<<std::endl;
+	    }
+
+	  }
+	  else
+	  {
+	    std::string warning = "You asked for QoI error estimates but did not use an Adjoint Refinement Error Estimator!\n";
+	    warning += "Please use the ADJOINT_REFINEMENT option for the estimator_type if you want QoI error estimates.\n";
+	    grins_warning(warning);
```
Comment (pbauman) [inline src/solver/src/steady_mesh_adaptive_solver.C]: Is there any way we could figure this out at construction time and throw an error instead? That would be nicer for the user so they don't have to wait for a solve if they mucked up their choice.

**Comment by pbauman** [comment]
Looks like history is clean now so you can just push commits directly to the branch and they'll show up here.

**Comment by vikramvgarg** [comment]
How can I access the error_estimator object from mesh_adaptive_solver_base ?

On Sun, Nov 1, 2015 at 1:51 PM, Paul T. Bauman <notifications@github.com>
wrote:

> Looks like history is clean now so you can just push commits directly to
> the branch and they'll show up here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grinsfem/grins/pull/326#issuecomment-152857685>.
>



-- 
Vikram Garg
Postdoctoral Associate
Predictive Engineering and Computational Science (PECOS)
The University of Texas at Austin
http://web.mit.edu/vikramvg/www/

http://vikramvgarg.wordpress.com/
http://www.runforindia.org/runners/vikramg

**Comment by vikramvgarg** [comment]
Is it okay if I do the check for ARefEE at the beginning
of SteadyMeshAdaptiveSolver::solve ?

On Sun, Nov 1, 2015 at 2:43 PM, Vikram Garg <vikram.v.garg@gmail.com> wrote:

> How can I access the error_estimator object from mesh_adaptive_solver_base
> ?
>
> On Sun, Nov 1, 2015 at 1:51 PM, Paul T. Bauman <notifications@github.com>
> wrote:
>
>> Looks like history is clean now so you can just push commits directly to
>> the branch and they'll show up here.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/grinsfem/grins/pull/326#issuecomment-152857685>.
>>
>
>
>
> --
> Vikram Garg
> Postdoctoral Associate
> Predictive Engineering and Computational Science (PECOS)
> The University of Texas at Austin
> http://web.mit.edu/vikramvg/www/
>
> http://vikramvgarg.wordpress.com/
> http://www.runforindia.org/runners/vikramg
>



-- 
Vikram Garg
Postdoctoral Associate
Predictive Engineering and Computational Science (PECOS)
The University of Texas at Austin
http://web.mit.edu/vikramvg/www/

http://vikramvgarg.wordpress.com/
http://www.runforindia.org/runners/vikramg

**Comment by pbauman** [comment]
> Is it okay if I do the check for ARefEE at the beginning
of SteadyMeshAdaptiveSolver::solve ?

That works for me.

**Comment by pbauman** [comment]
Actually, make it a "helper function" and call the helper function at the beginning of the solve.

**Comment by vikramvgarg** [comment]
Made relevant changes, can you check if its ready for merge now ?

**Comment by pbauman** [comment]
That last fix and this should be good.

**Comment by vikramvgarg** [comment]
Done.

File: src/solver/src/mesh_adaptive_solver_base.C [hunk 3]
```diff
@@ -45,6 +45,7 @@ namespace GRINS
       _refine_fraction( input("MeshAdaptivity/refine_percentage", 0.2) ),
       _coarsen_fraction( input("MeshAdaptivity/coarsen_percentage", 0.2) ),
       _coarsen_threshold( input("MeshAdaptivity/coarsen_threshold", 0) ),
+      _compute_qoi_error_estimate( input("MeshAdaptivity/compute_QoI_error_estimate", false)),
```
Comment (pbauman) [inline src/solver/src/mesh_adaptive_solver_base.C]: The input option too, please. QoI -> qoi

**Comment by vikramvgarg** [comment]
Done.

**Comment by pbauman** [comment]
Thanks!

**Pull request merged**